src/backend/distributed/utils/statistics_collection.c

```diff
@@ -112,46 +133,53 @@ CollectBasicUsageStatistics(void)
 	appendStringInfo(fields, ",\"node_metadata\": %s", metadataJsonbStr);
 	appendStringInfoString(fields, "}");
 
-	CommitTransactionCommand();
-
 	return SendHttpPostJsonRequest(STATS_COLLECTION_HOST "/v1/usage_reports",
 								   fields->data, HTTP_TIMEOUT_SECONDS);
 }
 
 
 /*
- * ClusterSize returns total size of data store in the cluster consisting of
- * given distributed tables. We ignore tables which we cannot get their size.
+ * DistributedTablesSize returns total size of data store in the cluster consisting
+ * of given distributed tables. We ignore tables which we cannot get their size.
  */
 static uint64
-ClusterSize(List *distributedTableList)
+DistributedTablesSize(List *distTableOids)
 {
-	uint64 clusterSize = 0;
-	ListCell *distTableCacheEntryCell = NULL;
+	uint64 totalSize = 0;
+	ListCell *distTableOidCell = NULL;
 
-	foreach(distTableCacheEntryCell, distributedTableList)
+	foreach(distTableOidCell, distTableOids)
 	{
-		DistTableCacheEntry *distTableCacheEntry = lfirst(distTableCacheEntryCell);
-		Oid relationId = distTableCacheEntry->relationId;
-		MemoryContext savedContext = CurrentMemoryContext;
-
-		PG_TRY();
+		Oid relationId = lfirst_oid(distTableOidCell);
+
+		/*
+		 * Relations can get dropped after getting the Oid list and before we
+		 * reach here. Acquire a lock to make sure the relation is available
+		 * while we are getting its size.
+		 */
+		Relation relation = try_relation_open(relationId, AccessShareLock);
+		if (relation == NULL)
 		{
-			Datum distTableSizeDatum = DirectFunctionCall1(citus_table_size,
-														   ObjectIdGetDatum(relationId));
-			clusterSize += DatumGetInt64(distTableSizeDatum);
+			continue;
 		}
-		PG_CATCH();
-		{
-			FlushErrorState();
 
-			/* citus_table_size() throws an error while the memory context is changed */
-			MemoryContextSwitchTo(savedContext);
+		/*
+		 * Ignore hash partitioned tables with size greater than 1, since
+		 * citus_table_size() doesn't work on them.
```

Comment:
Should we open an issue to track this? Seems like a shortcoming, especially for more legacy-ish users.

Reply:
Done at #1735.